Sources/Amplitude/Amplitude.m

```diff
@@ -1686,7 +1686,7 @@ - (void)initializeDeviceInfo {
         if (self.deviceId == nil) {
             self.deviceId = [self.dbHelper getValue:DEVICE_ID];
             if (![self isValidDeviceId:self.deviceId]) {
-                self.deviceId = [self _getDeviceId];
+                self->_deviceId = [self _getDeviceId];
```

Comment:
Actually not sure why this fixed the issue. What is the difference between assigning using `->` vs `.`?